specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json

```
@@ -0,0 +1,248 @@
+          "maxLength": 1024,
+          "type": "string"
+        },
+        "trunks": {
+          "description": "Gets or sets list of SIP trunks for routing calls. Trunks are represented as FQDN.",
+          "maxItems": 250,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SipConfiguration": {
+      "description": "Represents a SIP configuration.\r\nWhen a call is being routed the routes are applied in the same order as in the routes list.\r\nA route is matched by its number pattern.\r\nCall is then directed into route's first available trunk, based on the order in the route's trunks list.",
+      "type": "object",
+      "properties": {
+        "trunks": {
+          "description": "SIP trunks for routing calls.\r\nMap key is trunk's FQDN (1-249 characters).",
+          "maxProperties": 250,
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/Trunk"
+          }
+        },
+        "routes": {
+          "description": "Trunk routes for routing calls.",
+          "maxItems": 250,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TrunkRoute"
+          }
+        }
+      }
+    },
+    "SipConfigurationPatch": {
+      "description": "Represents a SIP configuration patch.",
+      "type": "object",
+      "properties": {
+        "trunks": {
+          "description": "SIP trunks for routing calls.\r\nMap key is trunk's FQDN (1-249 characters).",
+          "maxProperties": 250,
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/TrunkPatch"
+          }
+        },
+        "routes": {
+          "description": "Trunk routes for routing calls.",
+          "maxItems": 250,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TrunkRoute"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "in": "query",
+      "name": "api-version",
+      "description": "Version of API to invoke",
+      "required": true,
+      "type": "string"
+    },
+    "Endpoint": {
+      "in": "path",
+      "name": "endpoint",
+      "description": "The communication resource, for example https://resourcename.communication.azure.com",
+      "required": true,
+      "type": "string",
+      "x-ms-skip-url-encoding": true,
+      "x-ms-parameter-location": "client"
+    }
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "flow": "implicit",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      },
+      "description": "Azure Active Directory OAuth2 Flow"
+    }
+  },
```

You should specify `security` either globally or on each operation to associate the "azure_auth" security definition with each operation.
```suggestion
  },
  "security": [
    {
      "azure_auth": [ "user_impersonation" ]
    }
  ],
```

---

Thank you @mikekistler , I added the suggested security property. We as well added HMAC authentication that's supported but was omitted in the definition